Python3 unit test failure: //tensorflow/contrib/layers:summaries_test

This failure occurs under Python3, but not Python2. 
`bazel test -c opt //tensorflow/contrib/layers:summaries_test`

Other environment info: 
OS: "Linux", kernel: "4.2.0-18-generic", architecture: "x86_64", processor: "Intel(R) Xeon(R) CPU @ 2.50GHz", processor_count: "32", memory_total: "61845920 kB", swap_total: "0 kB", Bazel_version: "Build label: 0.1.4", Java_version: "1.8.0_72-internal", pp_version: "g++ (Ubuntu 4.8.4-2ubuntu1~14.04) 4.8.4"

> ==================== Test output for //tensorflow/contrib/layers:summaries_test:
> F./var/lib/jenkins/workspace/tensorflow-master-python3/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/68a62076e91007a7908bc42a32e4cff9/tensorflow/bazel-out/local_linux-py3-opt/bin/tensorflow/contrib/layers/summaries_test.runfiles/tensorflow/contrib/layers/python/layers/summaries_test.py:57: DeprecationWarning: Please use assertEqual instead.
>   self.assertEquals(len(names), 1)
> ......
> ======================================================================
> FAIL: test_duplicate_tag (__main__.SummariesTest)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/var/lib/jenkins/workspace/tensorflow-master-python3/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/68a62076e91007a7908bc42a32e4cff9/tensorflow/bazel-out/local_linux-py3-opt/bin/tensorflow/contrib/layers/summaries_test.runfiles/tensorflow/contrib/layers/python/layers/summaries_test.py", line 35, in test_duplicate_tag
>     tf.contrib.layers.summarize_tensor(var)
> AssertionError: ValueError not raised
> 
> ----------------------------------------------------------------------
> Ran 8 tests in 0.058s
> 
> FAILED (failures=1)

--- a/tensorflow/contrib/layers/python/layers/summaries.py
+++ b/tensorflow/contrib/layers/python/layers/summaries.py
@@ -35,7 +35,7 @@ __all__ = ['summarize_tensor', 'summarize_activation', 'summarize_tensors',
 def _assert_summary_tag_unique(tag):
   for summary in ops.get_collection(ops.GraphKeys.SUMMARIES):
     old_tag = tensor_util.constant_value(summary.op.inputs[0])
-    if tag == str(old_tag):
+    if tag.encode() == old_tag:
       raise ValueError('Conflict with summary tag: %s exists on summary %s %s' %
                        (tag, summary, old_tag))
 

